onnxruntime/core/providers/cpu/tensor/cast_op.cc

```diff
@@ -170,4 +175,57 @@ Status Cast<MLFloat16>::Compute(OpKernelContext* context) const {
   return st;
 }
 
+ONNX_CPU_OPERATOR_TYPED_KERNEL(
+    Cast,
+    9,
+    string,
+    KernelDefBuilder().TypeConstraint("T1", DataTypeImpl::GetTensorType<std::string>()).TypeConstraint("T2", castOpTypeConstraints),
+    Cast<std::string>);
+
+template <>
+Status Cast<std::string>::Compute(OpKernelContext* context) const {
+  const Tensor* X = context->Input<Tensor>(0);
+  if (X == nullptr) return Status(common::ONNXRUNTIME, common::FAIL, "input count mismatch");
+  const TensorShape& shape = X->Shape();
```

Comment:
I believe the ONNX validation would have rejected this much earlier if there was no input, so this probably isn't necessary or could be ORT_ENFORCE as it's unexpected.

Reply:
I agree with the validation part. I was keep the code aligning how we implemented existing kernel in the same file.

Reply:
Would it be better to make them both more correct instead of aligning with something that has unnecessary code?

---
In reply to: [251093319](https://github.com/Microsoft/onnxruntime/pull/379#discussion_r251093319) [](ancestors = 251093319)

Reply:
What if the input is a dictionary or a sequence? I am not sure if this line is useful for detecting type-mismatch.

---
In reply to: [251591057](https://github.com/Microsoft/onnxruntime/pull/379#discussion_r251591057) [](ancestors = 251591057,251093319)